analytics.py

```diff
@@ -0,0 +1,110 @@
+import math
+import random
+from .utils import euclidean_distance
+from .utils import generate_random
+

```

Comment:
These import are great.  In general, you can omit the `.` notation.  Using the `from X import Y` notation can keep the namespace cleaner, but beware an error like:

```bash
  File "/home/travis/build/Geospatial-Python/assignment_06/analytics.py", line 89, in permutations
    permutations.append(average_nearest_neighbor_distance(utils.generate_random(n), marks))
NameError: name 'utils' is not defined
```
In your code (line 124) you are still tying to access in the `utils` namespace.